compiler-access/src/main/scala/org/alephium/ralph/lsp/access/compiler/RalphParserExtension.scala

```diff
@@ -1,13 +1,18 @@
 package org.alephium.ralph.lsp.access.compiler
 
 import fastparse._
-import org.alephium.ralph.SourceIndex
-import org.alephium.ralph.StatefulParser.{rawContract, rawInterface, rawTxScript, whitespace}
+import org.alephium.ralph.{Parser, SourceIndex}
+import org.alephium.ralph.SourceFileStatefulParser
 import org.alephium.ralph.lsp.access.compiler.ast.Tree
 import org.alephium.ralph.lsp.access.compiler.message.SourceIndexExtra._
+import com.typesafe.scalalogging.StrictLogging
 
 /** Functions that extend ralphc's default parser */
-object RalphParserExtension {
+class RalphParserExtension(file:java.net.URI) extends StrictLogging{
+    implicit val fileURI:java.net.URI = file
+
+    val statefulParser = new SourceFileStatefulParser()(Some(fileURI))
+    import statefulParser._
```

Comment:
`SourceFileStatefulParser` 👍🏼 .

Just minor adjustment: We can remove `implicit fileURI` and just pass it a function parameter where needed.

Implemented this in PR #131.